packages/sn-filter-pane/src/components/ListboxPopoverContainer.tsx

```diff
@@ -76,8 +97,125 @@ export const ListboxPopoverContainer = ({ resources, stores }: FoldedPopoverProp
+    const target = event.target as HTMLElement;
+    if (target.tagName === 'INPUT') {
+      return;
+    }
+    switch (event.key) {
+      case KEYS.ESC:
+        onEscape(event);
+        break;
+      case KEYS.ENTER:
+        // @ts-ignore
+        handleOpen({ event });
+        setTimeout(changeFocus, 500);
+        break;
+      case KEYS.LEFT:
+      case KEYS.RIGHT:
+        return; // let it propagate to top-level
+      default:
+        return;
+    }
+
+    // Note: We should not stop propagation here as it will block the containing app
+    // from handling keydown events.
+    event.preventDefault();
+  };
+
+  const onFoldedListboxDropdownEnter = (event: React.KeyboardEvent) => {
+    const target = event.target as HTMLElement;
+    if (target.classList.contains('folded-listbox')) {
+      setTimeout(changeFocus, 500);
+      return;
+    }
+    const firstFoldedListbox: HTMLElement = target.querySelector('.folded-listbox') as HTMLElement;
+    target.setAttribute('tabIndex', '-1');
+    target.blur();
+    firstFoldedListbox?.setAttribute('tabIndex', '0');
+    firstFoldedListbox?.focus();
+  };
+
+  const findIndex = (element: EventTarget, nodeList: NodeList) => {
+    for (let i = 0; i < nodeList.length; i++) {
+      if (element === nodeList.item(i)) return i;
+    }
+    return -1;
+  };
+
+  const moveToNextFoldedListboxItem = (event: React.KeyboardEvent) => {
+    const target = event.target as HTMLElement;
+    const dropdown = target.closest('.folded-listbox-dropdown');
+    const foldedListboxList = dropdown?.querySelectorAll('.folded-listbox');
+    if (foldedListboxList?.length) {
+      const activeIndex = findIndex(event.target, foldedListboxList);
+      if (activeIndex < 0) {
+        return;
+      }
+      const nextIndex = event.key === KEYS.UP ? activeIndex - 1 : activeIndex + 1;
+      if (nextIndex < 0 || nextIndex >= foldedListboxList.length) {
+        return;
+      }
+      const elementToFocus = foldedListboxList.item(nextIndex);
+      if (elementToFocus) {
+        target?.setAttribute('tabIndex', '-1');
+        target?.blur();
+        elementToFocus.setAttribute('tabIndex', '0');
+        (elementToFocus as HTMLElement).focus();
+      }
+    }
+
+    const firstFoldedListbox: HTMLElement = target.querySelector('.folded-listbox') as HTMLElement;
+    target.setAttribute('tabIndex', '-1');
+    target.blur();
+    firstFoldedListbox?.setAttribute('tabIndex', '0');
+    firstFoldedListbox?.focus();
+  };
+
+  const onFoldedListboxDropdownKeyDown = (event: React.KeyboardEvent) => {
+    const target = event.target as HTMLElement;
+    if (target.tagName === 'INPUT') {
+      return;
+    }
+    switch (event.key) {
+      case KEYS.ESC:
+        onEscape(event);
+        handleClose();
+        break;
+      case KEYS.ENTER:
+        onFoldedListboxDropdownEnter(event);
+        break;
+      case KEYS.UP:
+      case KEYS.DOWN:
```

Comment:
Question: Should we also support left and right?
```
case KEYS.LEFT:
case KEYS.RIGHT:
```

Reply:
This is dropdown so LEFT and RIGHT is not needed. But maybe.

Reply:
maybe for screen reader user. they don't see it and don't know if it is a up/down or left/right navigation.
supporting both could make it easier for them?